File: app/src/main/java/io/rousan/androidwithrust/message/What.java
package io.rousan.androidwithrust.message;

public final class What {
    public final static int SERVER_STARTED = 0;
    public final static int INIT_DATA = 1;
    public final static int SEND_FILE = 2;
    public final static int SEND_FILE_PROGRESS = 3;
    public final static int SEND_FILE_DONE = 4;

    public static String toString(int what) {
        switch (what) {
            case SERVER_STARTED: {
                return "SERVER_STARTED";
            }
            case INIT_DATA: {
                return "INIT_DATA";
            }
            case SEND_FILE: {
                return "SEND_FILE";
            }
            case SEND_FILE_PROGRESS: {
                return "SEND_FILE_PROGRESS";
            }
            case SEND_FILE_DONE: {
                return "SEND_FILE_DONE";
            }
            default: {
                return "<UNKNOWN>";
            }
        }
    }
}

File: app/src/main/rust/message/what.rs
pub const SERVER_STARTED: i32 = 0;
pub const INIT_DATA: i32 = 1;
pub const SEND_FILE: i32 = 2;
pub const SEND_FILE_PROGRESS: i32 = 3;
pub const SEND_FILE_DONE: i32 = 4;

pub fn to_string(what: i32) -> &'static str {
    match what {
        SERVER_STARTED => "SERVER_STARTED",
        INIT_DATA => "INIT_DATA",
        SEND_FILE => "SEND_FILE",
        SEND_FILE_PROGRESS => "SEND_FILE_PROGRESS",
        SEND_FILE_DONE => "SEND_FILE_DONE",
        _ => "<UNKNOWN>",
    }
}
